std/encoding/base64.ts

```diff
@@ -9,13 +9,16 @@ const base64abc = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L",
 
 /**
  * CREDIT: https://gist.github.com/enepomnyaschih/72c423f727d395eeaa09697058238727
- * Encodes a given Uint8Array or string into RFC4648 base64 representation
+ * Encodes a given Uint8Array, ArrayBuffer or string into RFC4648 base64 representation
  * @param data
  */
-export function encode(data: Uint8Array | string): string {
-  const uint8 = typeof data === "string"
-    ? new TextEncoder().encode(data)
-    : data;
+export function encode(data: Uint8Array | ArrayBuffer | string): string {
```

Comment:
I would be fine with dropping the `ArrayBuffer` as the other APIs don't support it, but `Uint8Array` is assignable to `ArrayBuffer` (all typed arrays are also `ArrayBuffer`s).  [You can see from the following code, just having `ArrayBuffer` is actually valid](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXxHR1AAooAjALngEEYYoBPAIWUURBngB94BnDDCyoA5gEpqAoaIDcAKDlg8A+MgAc8ALzxUIAO7wAqsIxq6DRiTHy5hJaXXW5QA).  Your coercion to Uint8Array though is good:

```ts
declare function encode(ab: ArrayBuffer | string): string;

const u8 = new Uint8Array();

encode(u8);
```

The problem is that all the other encodes don't use `ArrayBuffer` (I would argue they should) and it clearly confused some people, not realising a `Uint8Array` is assignable to an `ArrayBuffer` (and not supporting coercing other typed arrays to `Uint8Array`.

Reply:
Thank you for correcting me. I just fixed it.
 
I think you were right with regards to `ArrayBuffer` anyways because as you can see my commit `adc874e` (where I removed `ArrayBuffer`) failed one test in `/node/string_decoder_test.ts`. I was not able to fix it without changing a lot of code in `string_decoder.ts`.